drivers/sensor/lis2dw12/lis2dw12.c

```diff
@@ -122,6 +137,24 @@ static inline void lis2dw12_channel_get_acc(const struct device *dev,
 	}
 }
 
+static int lis2dw12_get_die_temp(const struct device *dev, enum sensor_channel chan,
+				     struct sensor_value *val)
+{
+	int rc;
+	const struct lis2dw12_device_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	rc = lis2dw12_temperature_raw_get(ctx, (int16_t *)val);
```

Comment:
Please use a dedicated variable for temperature reading instead of trying to reuse `val`. You do not want to be dependent on the structure of sensor_value.